### PR TITLE
Fix normalization layers

### LINKED ISSUE #46366: LayerNormalization crashes on empty inputs when run on CPU
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Ubuntu 18.04
- TensorFlow installed from: binary
- TensorFlow version: 2.4.0
- Python version: 3.6.9

**Describe the current behavior**

`tf.keras.layers.LayerNormalization` crashes when the input is empty and the layer is executed on CPU.

**Describe the expected behavior**

The layer should not crash but return a tensor with the same shape.

**Standalone code to reproduce the issue**

```python
import os
os.environ["CUDA_VISIBLE_DEVICES"] = ""
import tensorflow as tf
layer = tf.keras.layers.LayerNormalization()
layer(tf.zeros([1, 0, 10]))
```

**Other info / logs**

The code above exits with this error:

```text
Floating point exception (core dumped)
```

### PR DESCRIPTION
**Pull request by around-star**
Fix #46366 . `LayerNormalization` crashes on getting an empty input while `BatchNormalization` returns back the empty input. This PR will help both the normlaization layers to raise an error upon encountering an empty input.

### CODE REVIEW

**Comment by around-star** [comment]
@deeb02  can you please review this PR?

File: tensorflow/python/keras/layers/normalization.py [hunk 1]
```diff
@@ -750,7 +750,10 @@ def _get_training_value(self, training=None):
 
   def call(self, inputs, training=None):
     training = self._get_training_value(training)
```
Comment (walli55) [inline tensorflow/python/keras/layers/normalization.py:752]: Alomaisi

Comment (walli55) [reply tensorflow/python/keras/layers/normalization.py:752]: Alomaisi

**Review by walli55 — APPROVED** [review]

**Comment by around-star** [comment]
@mattdangerw can you please review this PR?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR.

`inputs.shape` is the static shape, but in practice you'd want to check `tf.shape(inputs)`.

However, adding such a check on the critical path will create an unnecessary slowdown for the 99.999% of users who do not call the layer with an input with a 0 dimension. Besides, it's likely that adding the error for something that was previously allowed will break a small number of existing users.

We shouldn't do any such check.
